### PR TITLE
Nav redesign / jetpack mu - Unregister customizer items when on block theme.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
Fixes # https://github.com/Automattic/dotcom-forge/issues/5853 

## Proposed changes:

Removes the 'customize' menu and adminbar link from atomic sites on block themes running the nav redesign.

* This is essentially a copy of @okmttdhr's work on https://github.com/Automattic/jetpack/pull/36017
* The above linked fix is in the masterbar module, which is disabled as part of the new nav redesign.
* This PR ports the same code to jetpack-mu-wpcom plugin, and applies them to atomic sites. 

BEFORE

Screenshot 2024-03-04 at 2 13 10 PM

Screenshot 2024-03-04 at 2 13 00 PM

AFTER

Screenshot 2024-03-04 at 2 12 26 PM
Screenshot 2024-03-04 at 2 12 34 PM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* On a WOA dev blog, enable the "Classic" interface setting under hosting->configuration settings.
* Be proxied, or add the `wpcom_classic_early_release` site option to your site.
* Ensure you have a block theme enabled.
* Build and sync this branch of jetpack-mu-wpcom. PCYsg-Osp-p2
* Verify the "Customizer" link no longer appears under:
   *  "Appearance" in the wp-admin left nav menu
   * In the top adminbar while viewing the front end of the site.
* visit the customizer directly by URL `*/wp-admin/customize.php`, verify this loads as expected without errors.

* Enable a classic theme and verify the customize menu reappears.

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php [hunk 1]
```diff
@@ -292,3 +292,28 @@ function wpcom_site_menu_handle_dismiss_notice() {
 	wp_die();
 }
 add_action( 'wp_ajax_dismiss_wpcom_site_menu_intro_notice', 'wpcom_site_menu_handle_dismiss_notice' );
+
+/**
+ * Ensures customizer menu and adminbar items are not visible on a block theme for atomic sites.
+ */
+function hide_customizer_menu_on_block_theme() {
+	$is_wpcom = ( defined( 'IS_WPCOM' ) && IS_WPCOM );
+	if ( ! $is_wpcom && wp_is_block_theme() ) {
```
Comment (okmttdhr) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: Could you add the `! is_customize_preview()` condition to give access to the features if users visit the Customizer directly? 🙂  (c.f., https://github.com/Automattic/jetpack/pull/36146)

```suggestion
	if ( ! $is_wpcom && wp_is_block_theme() && ! is_customize_preview() ) {
```

Comment (Addison-Stavlo) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-site-menu/wpcom-site-menu.php]: Thank you! Committed 😁 

Noting, without this change there weere errors in the customizer when visiting the url directly. This addition resolves that.

**Comment by Addison-Stavlo** [comment]
This is ready for another look.

**Comment by Addison-Stavlo** [comment]
> WordPress.com Tests Failing after 3m — Failed! D140934-code

I reran this on the diff and it seems to be good now.

**Review by okmttdhr — APPROVED** [review]
Confirmed "Customize" is hidden for Block themes and shown for Classic themes. LGTM 👍

**Pull request merged**